lib/puppet/resource_api/base_context.rb

```diff
@@ -19,13 +24,8 @@ def failed?
   end
 
   def feature_support?(feature)
-    supported = Puppet::Type.type(@typename).feature_support?(feature)
-    if supported
-      Puppet.debug("#{@typename} supports `#{feature}`")
-    else
-      Puppet.debug("#{@typename} does not support `#{feature}`")
-    end
-    supported
+    Puppet.info('DEPRECATED: Please use context.type.feature? instead.')
```

Comment:
Could use puppet's built-in `deprecation_warning(msg, key)` function: https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/lib/puppet/functions/deprecation.rb#L28
https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/logging.rb#L129